src/gt4py/next/program_processors/runners/dace_fieldview/program.py

```diff
@@ -0,0 +1,251 @@
+from gt4py.next.ffront import decorator
+from gt4py.next.iterator import ir as itir
+from gt4py.next.otf import arguments, recipes, toolchain
+from gt4py.next.program_processors.runners.dace_common import utility as dace_utils
+from gt4py.next.type_system import type_specifications as ts
+
+
+@dataclasses.dataclass(frozen=True)
+class Program(decorator.Program, dace.frontend.python.common.SDFGConvertible):
+    """Extension of GT4Py Program implementing the SDFGConvertible interface via GTIR."""
+
+    sdfg_closure_cache: dict[str, Any] = dataclasses.field(default_factory=dict)
+    # Being a ClassVar ensures that in an SDFG with multiple nested GT4Py Programs,
+    # there is no name mangling of the connectivity tables used across the nested SDFGs
+    # since they share the same memory address.
+    connectivity_tables_data_descriptors: ClassVar[
+        dict[str, dace.data.Array]
+    ] = {}  # symbolically defined
+
+    def __sdfg__(self, *args: Any, **kwargs: Any) -> dace.sdfg.sdfg.SDFG:
+        if (self.backend is None) or "dace" not in self.backend.name.lower():
+            raise ValueError("The SDFG can be generated only for the DaCe backend.")
+
+        offset_provider = (self.connectivities or {}) | self._implicit_offset_provider
+        column_axis = kwargs.get("column_axis", None)
+
+        gtir_stage = typing.cast(next_backend.Transforms, self.backend.transforms).past_to_itir(
+            toolchain.CompilableProgram(
+                data=self.past_stage,
+                args=arguments.CompileTimeArgs(
+                    args=tuple(p.type for p in self.past_stage.past_node.params),
+                    kwargs={},
+                    column_axis=column_axis,
+                    offset_provider=offset_provider,
+                ),
+            )
+        )
+
+        compile_workflow = typing.cast(
+            recipes.OTFCompileWorkflow,
+            self.backend.executor
+            if not hasattr(self.backend.executor, "step")
+            else self.backend.executor.step,
+        )  # We know which backend we are using, but we don't know if the compile workflow is cached.
+        sdfg = dace.SDFG.from_json(compile_workflow.translation(gtir_stage).source_code)
+
+        self.sdfg_closure_cache["arrays"] = sdfg.arrays
+
+        # Halo exchange related metadata, i.e. gt4py_program_input_fields, gt4py_program_output_fields, offset_providers_per_input_field
+        # Add them as dynamic properties to the SDFG
+        program = typing.cast(
+            itir.Program, gtir_stage.data
+        )  # we already checked that our backend uses GTIR
+        field_params = {
+            str(param.id): param for param in program.params if isinstance(param.type, ts.FieldType)
+        }
+
+        input_fields = (field_params[name] for name in InputNamesExtractor.only_fields(program))
+        # TODO (ricoh): This will associate the last horizontal dimension of each field with it's name
```

Comment:
My suspect is that this works under the assumption that all fields have a single horizontal dimension.
@kotsaloscv If this is the case, should we make a check and raise an exception if this assumption is not met?

Reply:
We had discussed this and I refactored it to explicitly only add fields with a single horizontal dimension.